### ISSUE TITLE
Error: $injector:modulerr Module Error

### ISSUE THREAD

**Issue by RayKahn** [Chevtek/angular-spinners#23]
Hi,
I am new to AngularJS but would like to use you spinner service. I have followed your instruction but I am getting the error mentioned in the title. My set us is as follow:
`app.js`
`(function () {`
   `angular.module('medmart', [`
       `'ui.router',`  
       `'oc.lazyLoad',                  // ocLazyLoad`
       `'ui.bootstrap',                 // Ui Bootstrap`
       `'ngFileUpload',`
       `'ui.validate',`
      `'angularSpinners'', // I removed I am not getting the exception`
       `'btford.socket-io',`
  `]).`
    `factory('mySocket', function (socketFactory) {`
        `var mySocket = socketFactory({`
           `ioSocket: io.connect(_URL),`
       `});`
        `mySocket.forward('error');`
       `mySocket.forward('broadcast');`
        `return mySocket;`
    `});`
`})();`

In my controller.js I have the follwoing:

`function editProviderWizardCtrl($scope, $rootScope, $stateParams, $http, $location, Upload, SweetAlert, spinnerService) {`
`$scop.spinnerService    = spinnerService;`
`... code here`
`}`

I have placed the anular-spinners.min.js in 'js/plugins/spinners/angular-spinners.min.js'

Can you tell me what the issue could be.

Thanks 
-r

**Comment by AlexAllocated** [comment]
I'm sorry this is really hard to read and nothing jumps out at me right away.

Pro tip: You don't have to wrap each line in backtick characters to show code here. You can put three backticks above your code and three more at the bottom. Like this:

``````
\```
  if (stuff) {
    doStuff();
  }
\```
``````

Not including the backslashes though. I had to put those there so it wouldn't interpret the characters. You can also add a language after the first three backticks to get syntax highlighting in your code snippet.

`````` javascript
\```javascript
  if (stuff) {
    doStuff();
  }
\```
``````

**Comment by skirscher** [comment]
At first look, I see two errors in the code posted (and thus your code if directly c&p'd).

The first:
```javascript
'angularSpinners'', // I removed I am not getting the exception
```
should be:
```javascript
'angularSpinners',
```

...

The second:
```javascript
$scop.spinnerService = spinnerService;
```

should be:
```javascript
$scope.spinnerService = spinnerService;
```

These were the only things that jumped out at me though. Hope this helps, I'm still kind of new to AngularJS myself but a veteran programmer overall.

**Comment by skirscher** [comment]
Funny, I just added the spinner to a project I'm working on and got the same error after trying to move the "spinnerService.hide('searchSpinner')" part outside of my "finally" clause to my http.get.then call. I even tried to move it back and still getting the same error. I only did this because when I originally loaded my page with the spinner, it showed it but didn't hide it once it was finished loading my data. Not sure where to go from here but would appreciate any insight you might have. Forgive me for not presenting the code via forum as I'm creating this project for a client but the jist of my code is I have my spinnerService.show('searchSpinner'); before my http.get.then which grabs my JSON data from my pl script, stores the result in scope & ends. Then in the finally clause afterwards, I have my hide call to the spinner service. Hope that helps to better understand the code without actually seeing it. If you absolutely must see it, I can possibly share snippets via private messaging.

**Comment by AlexAllocated** [comment]
Yeah it's gonna be real hard for me to know the exact problem without a snippet. Or better yet, if you can recreate your error in a [codepen](http://codepen.io] then it would be even easier for me to help you debug. If you prefer to email you can send snippets to alex.ford@codetunnel.com.

**Comment by skirscher** [comment]
So I realized I copied your code from your article and not the github version; copied the github version over my previous copy and the injector error went away. However, now I have an error saying my spinner isn't registered to the service. If it's not one thing it's another right? I'm working on reproducing the error in a plunker now...wish me luck that I can actually duplicate it.

**Comment by skirscher** [comment]
In attempting to re-create the error (which I sort of did here, at least the original error, not the "spinner" not registered one: http://plnkr.co/edit/vM1S9EoHyK3BB3LDatex?p=preview ; note: doing so was done rather hastely, not sure I had everything in the right place) I found that it doesn't seem to like to be included in nested views. Specifically ones using ui-router. My plunker does this and got the original error I was having and when I took the spinner tag out of my nested html page and put it into my index.html where the primary ui-view container is (and even put it inside the container itself), the errors were completely resolved yet the loading spinner didn't appear. Keeps on getting curious-er and curious-er. Then again, I always end up with the strange situations that no one who helps me has ever seen before; I'm talented like that. 0.o ^_^
